test/new.jl

```diff
@@ -2294,7 +2294,7 @@ end
     end
 end
 
-tree_hash(root::AbstractString) = bytes2hex(Pkg.GitTools.tree_hash(root))
+tree_hash(root::AbstractString) = bytes2hex(@inferred Pkg.GitTools.tree_hash(root))
```

Comment:
Perhaps just making a separate test to test that it infers is slightly better than calling `@inferred` over and over (many times with the same argument).